code/datums/brain_damage/imaginary_friend.dm

```diff
@@ -153,25 +174,32 @@
 	to_chat(src, compose_message(speaker, message_language, raw_message, radio_freq, spans, message_mode))
 
 /mob/camera/imaginary_friend/proc/friend_talk(message)
-	message = capitalize(trim(copytext_char(sanitize(message), 1, MAX_MESSAGE_LEN)))
+	message = treat_message_min(trim(copytext_char(sanitize(message), 1, MAX_MESSAGE_LEN)))
 
 	if(!message)
 		return
 
 	src.log_talk(message, LOG_SAY, tag="imaginary friend")
 
+	// Display message
+	if (!owner.client?.prefs.chat_on_map)
+		var/mutable_appearance/MA = mutable_appearance('icons/mob/talk.dmi', src, "default[say_test(message)]", FLY_LAYER)
+		MA.appearance_flags = APPEARANCE_UI_IGNORE_ALPHA
+		INVOKE_ASYNC(GLOBAL_PROC, /proc/flick_overlay, MA, list(owner.client), 30)
+	if(owner.client?.prefs.chat_on_map || client?.prefs.chat_on_map)
+		var/list/hearers = list()
+		if(client?.prefs.chat_on_map)
+			hearers += client
+		if(owner.client?.prefs.chat_on_map)
+			hearers += owner.client
+		new /datum/chatmessage(message, src, hearers, null)
```

Comment:
Isn't there some wrapper proc you should call before new chatmessages? I'd have to check myself but I vaguely remember it existing

Reply:
yes but it basically does a bunch of logic for hearing and emotes that don't work in this case